CharlieBackend.Api.UnitTes/HomeworkServiceTests.cs

```diff
@@ -0,0 +1,293 @@
+        private readonly Mock<IHomeworkRepository> _homeworkRepositoryMock;
+        private readonly Mock<IAttachmentRepository> _attachmentRepositoryMock;
+        private readonly Mock<ILessonRepository> _lessonRepositoryMock;
+        private readonly HomeworkService _homeworkService;
+
+        public HomeworkServiceTests()
+        {
+            _loggerMock = new Mock<ILogger<HomeworkService>>();
+            _homeworkRepositoryMock = new Mock<IHomeworkRepository>();
+            _attachmentRepositoryMock = new Mock<IAttachmentRepository>();
+            _lessonRepositoryMock = new Mock<ILessonRepository>();
+            _mapper = GetMapper(new ModelMappingProfile());
+            _homeworkService = new HomeworkService(_unitOfWorkMock.Object, _mapper, _loggerMock.Object);
+        }
+
+        #region CreateHomeworkAsync
+
+        [Fact]
+        public async Task CreateHomeworkAsync_DefaultHomeworkRequestDto_ShouldReturnValidationError()
+        {
+            //Arrange
+            HomeworkRequestDto homeworkRequestDto = default;
+
+            //Act
+            var result = await _homeworkService.CreateHomeworkAsync(homeworkRequestDto);
+
+            //Assert
+            result.Error.Code.Should().BeEquivalentTo(ErrorCode.ValidationError);
+
+        }
+
+        [Fact]
+        public async Task CreateHomeworkAsync_NotExistingLessonId_ShouldReturnValidationError()
+        {
+            //Arrange
+            HomeworkRequestDto homeworkRequestDto = new HomeworkRequestDto()
+            {
+                LessonId = 1
+            };
+
+            _lessonRepositoryMock.Setup(l => l.IsEntityExistAsync(1)).ReturnsAsync(false);
+            _unitOfWorkMock.Setup(x => x.LessonRepository).Returns(_lessonRepositoryMock.Object);
+
+            //Act
+            var result = await _homeworkService.CreateHomeworkAsync(homeworkRequestDto);
+
+            //Assert
+            result.Error.Code.Should().BeEquivalentTo(ErrorCode.ValidationError);
+
+        }
+
+        [Fact]
+        public async Task CreateHomeworkAsync_NotExistingAttachment_ShouldReturnValidationError()
+        {
+            //Arrange
+            HomeworkRequestDto homeworkRequestDto = new HomeworkRequestDto()
+            {
+                LessonId = 1,
+                AttachmentIds = new List<long> { 1, 2 }
+            };
+
+            _lessonRepositoryMock.Setup(l => l.IsEntityExistAsync(1)).ReturnsAsync(false);
+            //_homeworkRepositoryMock.Setup(x => x.Add(It.IsAny<Homework>()))
+            //        .Callback<Homework>(x => new Homework { LessonId = homeworkRequestDto.LessonId });
+            _attachmentRepositoryMock.Setup(x => x.IsEntityExistAsync(It.IsAny<long>())).ReturnsAsync(false);
+
+            _unitOfWorkMock.Setup(x => x.LessonRepository).Returns(_lessonRepositoryMock.Object);
+            // _unitOfWorkMock.Setup(x => x.HomeworkRepository).Returns(_homeworkRepositoryMock.Object);
```

Comment:
Remove this commented code please, if you don't need it

Reply:
Done